pyobis/checklist/checklist.py

```diff
@@ -115,54 +128,56 @@ def redlist(
-        },
-        "application/json; charset=utf-8",
-        **kwargs
-    )
-    return out
-
-
-def newest(
-    scientificname=None,
-    taxonid=None,
-    nodeid=None,
-    startdate=None,
-    enddate=None,
-    startdepth=None,
-    enddepth=None,
-    geometry=None,
-    flags=None,
-    **kwargs
-):
-    """
-    Generate a checklist of most recently added species.
-
-    :param scientificname: [String] Scientific name. Leave empty to include
-        all taxa.
-    :param taxonid: [String] Taxon AphiaID.
-    :param nodeid: [String] Node UUID.
-    :param startdate: [String] Start date formatted as YYYY-MM-DD.
-    :param enddate: [String] End date formatted as YYYY-MM-DD.
-    :param startdepth: [Fixnum] Start depth, in meters. Depth below sea level are treated as
-        positive numbers.
-    :param enddepth: [Fixnum] End depth, in meters. Depth below sea level are treated as
-        positive numbers.
-    :param geometry: [String] Geometry, formatted as WKT or GeoHash.
-    :param flags: [String] Comma separated list of quality flags which need to
-        be set.
-
-    :return: A dictionary
-
-    Usage::
-
-       from pyobis import checklist as ch
-       ch.newest(scientificname='Abra Alba')
-    """
-    url = obis_baseurl + "checklist/newest"
-    scientificname = handle_arrstr(scientificname)
-    out = obis_GET(
-        url,
-        {
+        }
+        out = obis_GET(
+            OBISQueryResult.url,
+            OBISQueryResult.args,
+            "application/json; charset=utf-8",
+            **kwargs,
+        )
+        return out
+
+    def newest(
+        self,
+        scientificname=None,
+        taxonid=None,
+        nodeid=None,
+        startdate=None,
+        enddate=None,
+        startdepth=None,
+        enddepth=None,
+        geometry=None,
+        flags=None,
+        **kwargs,
+    ):
+        """
+        Generate a checklist of most recently added species.
+
+        :param scientificname: [String] Scientific name. Leave empty to include
+            all taxa.
+        :param taxonid: [String] Taxon AphiaID.
+        :param nodeid: [String] Node UUID.
+        :param startdate: [String] Start date formatted as YYYY-MM-DD.
+        :param enddate: [String] End date formatted as YYYY-MM-DD.
+        :param startdepth: [Fixnum] Start depth, in meters. Depth below sea level are treated as
+            positive numbers.
+        :param enddepth: [Fixnum] End depth, in meters. Depth below sea level are treated as
+            positive numbers.
+        :param geometry: [String] Geometry, formatted as WKT or GeoHash.
+        :param flags: [String] Comma separated list of quality flags which need to
+            be set.
+
+        :return: A dictionary
+
+        Usage::
+
+        from pyobis.checklist import ChecklistQuery as CQR
+        ch = CQR()
+        ch.newest(scientificname='Abra Alba')
+        """
+        OBISQueryResult.url = obis_baseurl + "checklist/newest"
```

Comment:
f-strings are much faster than string additions, and easier to read. Again, no need to change, just a minor suggestion.

```suggestion
        OBISQueryResult.url = f"{obis_baseurl}checklist/newest"
```

Reply:
That's a really great learning for me. Optimizing time is the thing I am reading more of in theoretical CS and I think your practical suggestions are helping me learn more. Thank you!